src/model/PropertyVerifier.h

```diff
@@ -0,0 +1,96 @@
+**
+** Copyright (C) 2018 Luxoft Sweden AB
+**
+** This file is part of the FaceLift project
+**
+** Permission is hereby granted, free of charge, to any person
+** obtaining a copy of this software and associated documentation files
+** (the "Software"), to deal in the Software without restriction,
+** including without limitation the rights to use, copy, modify, merge,
+** publish, distribute, sublicense, and/or sell copies of the Software,
+** and to permit persons to whom the Software is furnished to do so,
+** subject to the following conditions:
+**
+** The above copyright notice and this permission notice shall be
+** included in all copies or substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+** NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+** BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+** ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+** CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+** SOFTWARE.
+**
+** SPDX-License-Identifier: MIT
+**
+**********************************************************************/
+
+#pragma once
+
+#include <QString>
+#include <QObject>
+#include <functional>
+
+#include "FaceliftModel.h"
+
+namespace facelift {
+
+class FaceliftModelLib_EXPORT PropertyVerifierBase
+{
+public:
+    void onError();
+};
+
+/**
+ * This class can be used to verify the behavior of a property
+ */
+template<typename OwnerType, typename PropertyType>
+class PropertyVerifier : public PropertyVerifierBase
+{
+    typedef const PropertyType& (OwnerType::*GetterMethod)() const;
+    typedef void (OwnerType::*ChangeSignal)();
+
+public:
+
+    PropertyVerifier(OwnerType *service, GetterMethod getter, ChangeSignal changeSignal, const char *propertyName) : m_propertyName(propertyName)
+    {
+        m_service = service;
+
+        m_previousValue = (m_service->*getter)();
+        QObject::connect(m_service, changeSignal, m_service, [this, getter]() {
+            auto newValue = (m_service->*getter)();
+            if (newValue == m_previousValue) {
+                qCCritical(LogModel) << "Change signal triggered but the value has not changed" << newValue << m_service << "propertyName:" << m_propertyName;
+                this->onError();
+            }
+            m_previousValue = newValue;
+        });
+    }
+
+    OwnerType * m_service = nullptr;
+    PropertyType m_previousValue;
+    const char *m_propertyName;
+};
+
+template<typename OwnerType, typename PropertyType>
+static void createPropertyVerifier(const PropertyInterface<OwnerType, PropertyType>& property)
+{
+    new PropertyVerifier<OwnerType, PropertyType>(property.object(), property.getter(), property.signal(), property.name());
```

Comment:
Fixed

Reply:
It is OK now, but please add comment:
// memory managed by Qt ownership model
It may be confusing why there is operator new called this way if you read this first time.